xpublish_edr/geometry/common.py

```diff
@@ -19,3 +28,106 @@ def is_regular_xy_coords(ds: xr.Dataset) -> bool:
     Check if the dataset has 2D coordinates
     """
     return coord_is_regular(ds.cf["X"]) and coord_is_regular(ds.cf["Y"])
+
+
+def dataset_crs(ds: xr.Dataset) -> pyproj.CRS:
+    grid_mapping_names = ds.cf.grid_mapping_names
+    if len(grid_mapping_names) == 0:
+        # Default to WGS84
+        return pyproj.crs.CRS.from_epsg(4326)
+    if len(grid_mapping_names) > 1:
+        raise ValueError(f"Multiple grid mappings found: {grid_mapping_names!r}!")
+    (grid_mapping_var,) = tuple(itertools.chain(*ds.cf.grid_mapping_names.values()))
+
+    grid_mapping = ds[grid_mapping_var]
+    return pyproj.CRS.from_cf(grid_mapping.attrs)
+
+
+def project_geometry(ds: xr.Dataset, geometry_crs: str, geometry: Geometry) -> Geometry:
+    """
+    Get the projection from the dataset
+    """
+    data_crs = dataset_crs(ds)
+
+    transformer = transformer_from_crs(
+        crs_from=geometry_crs,
+        crs_to=data_crs,
+        always_xy=True,
+    )
+    return transform(transformer.transform, geometry)
+
+
+def project_dataset(ds: xr.Dataset, query_crs: str) -> xr.Dataset:
+    """
+    Project the dataset to the given CRS
+    """
+    data_crs = dataset_crs(ds)
+    target_crs = pyproj.CRS.from_string(query_crs)
+    if data_crs == target_crs:
+        return ds
+
+    transformer = transformer_from_crs(
+        crs_from=data_crs,
+        crs_to=target_crs,
+        always_xy=True,
+    )
+
+    # TODO: Handle rotated pole
```

Comment:
i thought we figured this out.

Reply:
This only matters if the user has input a rotated coord system as the target coordinate system. Which i may be comfortable ignoring for now...